components/camel-groovy/src/main/java/org/apache/camel/language/groovy/GroovyShellFactory.java

```diff
@@ -27,4 +29,14 @@ default String getFileName(Exchange exchange) {
         return null;
     }
 
+    /**
+     * This method provide a way to define some global variables that will be applied to all the groovy script context.
+     *
+     * @param exchange the camel exchange in process.
+     * @return the global variables that will be applied to all the groovy script context.
+     */
+    default Map<String, Object> getVariables(Exchange exchange) {
+        return null;
```

Comment:
Maybe we could return an empty map instead, to avoid having to check if it is null and then have a code easier to read